extending/provider/implementation.rst

```diff
@@ -111,35 +63,39 @@ The following example implementation of ``refresh()`` updates an EBS volume in A
          setState(volume.stateAsString());
          setVolumeType(volume.volumeTypeAsString());
 
-         Ec2Client client = createClient(Ec2Client.class);
-
          DescribeVolumeAttributeResponse responseAutoEnableIo = client.describeVolumeAttribute(
-             r -> r.volumeId(getId())
-                 .attribute(VolumeAttributeName.AUTO_ENABLE_IO)
+             r -> r.volumeId(getId()).attribute(VolumeAttributeName.AUTO_ENABLE_IO)
          );
 
          setAutoEnableIo(responseAutoEnableIo.autoEnableIO().value());
+
+         return true;
      }
 
-**create(GyroUI ui, State state)**
+Implementing create
+===================
+
+The ``create(GyroUI ui, State state)`` method is called to create a resource. Gyro will call this method for every resource
+it determines needs to be created.
 
-The ``create(..)`` method is called by Gyro when it determines that it should create a resource. Implementations should
-create the resource and update any unique ID fields on the current object instance that will be necessary to query for
-the resource ``refresh()`` method.
+Implementations should create the resource and update any output properties after creation. At a minimum the id
+that uniquely identifies the resource in the cloud provider should be set.
 
-Gyro will call ``create(..)`` if the resource does not exist in state or if a non-updatable field has been modified. In
-the later case Gyro will first call ``delete(..)``.
+.. note::
 
-The following example implementation of ``create(GyroUI ui, State state)`` creates an EBS volume in AWS:
+    Call ``state.save()`` in the create method after each API call that creates/updates the resource. This is necessary
+    when the cloud provider requires an API call to create the resource, then subsequent API calls to update
+    specific properties. If there is only one API call to create a resource then calling ``state.save()`` is not
```

Comment:
I would flip this, to have what _normally_ occurs first, then the exception as to when you would want to call save manually.